### PR TITLE
Feature/web darktheme - adding the basics for a dark theme toggle

### PR DESCRIPTION
**Pull request by meteiorman**
## Description

Adding a toggle for theme: dark/light/auto in the web UI
it sets a cookie so it can be set per device
Basic functionality seems to be in place but can be expanded further if needed
NOTE: This is my first ever pull request so PLEASE help me review and test the code to make sure I didn't break anything. I'm a rookie in here. Thanks!

### Screenshot

image

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src_assets/common/assets/web/scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/scripts/main.js:1]: Can you rename this directory to `js`?

Probably no blank line at start of file? Or add a comment describing the purpose of the script?

File: src_assets/common/assets/web/footer.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    server: '804382334370578482',
+    channel: '804383092822900797',
+    defer: false,
+  })
+</script>
+
+
+</body>
+
+</html>
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/footer.html:15]: Please add empty blank line at end of each file.

File: src_assets/common/assets/web/styles/main.css [hunk 3]
```diff
@@ -0,0 +1,35 @@
+.nav-link.active {
+  font-weight: 500;
+}
+
+
+
+
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/styles/main.css:7]: Can you make the blank lines between a consistent number? I'm not a css expert so don't know what the standard is. Probably 1 blank line?

File: src_assets/common/assets/web/scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    //set the theme & cookie
+    let theme = btn.getAttribute('data-bs-theme');
+    document.cookie = 'sunshineTheme='+theme+';';
+    
+    //remove all active classes from all buttons
+    btn.closest('.dropdown-menu').querySelectorAll('button').forEach((element) => {
+      element.classList.remove('active');
+    });
+    
+
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/scripts/main.js:35]: 1 blank line

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]
Thank you for this PR! I think it will be a great addition to Sunshine!

File: src_assets/common/assets/web/scripts/main.js (hunk 1 above)
Comment (meteiorman) [reply src_assets/common/assets/web/scripts/main.js:1]: Will do!

File: src_assets/common/assets/web/header.html [hunk 5]
```diff
@@ -1,4 +1,5 @@
 <!DOCTYPE html>
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (meteiorman) [reply src_assets/common/assets/web/header.html:2]: Good catch!

File: src_assets/common/assets/web/styles/main.css (hunk 3 above)
Comment (meteiorman) [reply src_assets/common/assets/web/styles/main.css:7]: Can do

File: src_assets/common/assets/web/footer.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<script src="/scripts/main.js"></script>
+
+<!-- Discord WidgetBot Crate-->
+<script src="https://cdn.jsdelivr.net/npm/@widgetbot/crate@3" async defer>
+  new Crate({
+    server: '804382334370578482',
+    channel: '804383092822900797',
+    defer: false,
+  })
+</script>
```
[the remark this answers is not in the document]
Comment (meteiorman) [reply src_assets/common/assets/web/footer.html:10]: I think I got what you are saying, please see my latest PR

File: src_assets/common/assets/web/footer.html (hunk 2 above)
Comment (ReenigneArcher) [reply src_assets/common/assets/web/footer.html]: For every file please... In GitHub it shows like this if it's not present.

image

File: src_assets/common/assets/web/footer.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<script src="/node_modules/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/footer.html:1]: Just a question... but is there a functional difference between these being here at the end of the html or in the `head` section as they were before?

Comment (meteiorman) [reply src_assets/common/assets/web/footer.html:1]: Good question, older practices would put scripts in the head and then you'd do some sort of an 'on ready' to run the scripts only after all the content is loaded. By putting them at the end, it's more modern practice and helps with the loading order. https://levelup.gitconnected.com/all-about-script-87fea475b976

**Review by ReenigneArcher — APPROVED** [review]

**Comment by ReenigneArcher** [comment]
I'm getting an error message when loading the welcome message.

```txt
Uncaught ReferenceError: Vue is not defined
    <anonymous> https://localhost:47990/welcome:75
welcome:75:3
```

**Comment by meteiorman** [comment]
> I'm getting an error message when loading the welcome message.
> 
> ```
> Uncaught ReferenceError: Vue is not defined
>     <anonymous> https://localhost:47990/welcome:75
> welcome:75:3
> ```

My bad! I didn't realize other pages had code within their body. In that case, can you move the vue.js script that's loaded in the footer.html up into the <head> again?

**Comment by ReenigneArcher** [comment]
> In that case, can you move the vue.js script that's loaded in the footer.html up into the (head) again?

Fine with me. Did you want me to make the change?

**Comment by meteiorman** [comment]
> > In that case, can you move the vue.js script that's loaded in the footer.html up into the (head) again?
> 
> Fine with me. Did you want me to make the change?

Sure! I'm not 100% how to make that change and get it to you correctly yet,😅

**Comment by ReenigneArcher** [comment]
I pushed a commit that allows us to get any asset from the `/web` directory. Still seeing some issues.

- Changing theme only seems to update the navbar color.
  - Dark theme has black/dark gray navbar, text stays dark so can't read text.
  - Light theme looks like before in navbar.
  
I didn't look much into the reason, but didn't see any errors in console.

**Comment by DanTheMan827** [comment]
A JavaScript toggle works, but most websites just use css media queries for dark/light mode if they have separate styles

**Comment by ReenigneArcher** [comment]
I'm not really sure what the issue could be. Here are some screenshots.

Auto:
image

Light:
image

Dark:
image

Some of the js/css is slow to load, but every resource does load after 1 or 2 seconds at least.

The head tag does have the `data-bs-theme` attribute changed. Looking at the docs (https://getbootstrap.com/docs/5.3/customize/color-modes/) it appears this feature was just added in v5.3. Your branch is using v5.2.x. I'm not sure why @dependabot hasn't tried to update this dependency yet, but I believe that is likely the reason it's not fully working.

**Comment by ReenigneArcher** [comment]
Updating w/ rebase to make this easier to fix conflicts after adjusting clang-format rules

**Comment by meteiorman** [comment]
Good catch, that's totally the issue - I'm basing it off of 5.3 - since you're using stock Bootstrap, I had pulled the latest from their page to get the color picker to work. I wasn't sure how those packages get updated so I didn't commit that latest version in my code. You could pause this until 5.3 is out of alpha, or go with alpha 3 for now and it'd probably work just fine as well.

**Comment by ReenigneArcher** [comment]
@meteiorman bootstrap has been updated to v5.3.0 now. Could you fix the conflicts in your PR and we can resume this development/testing?

**Comment by ReenigneArcher** [comment]
Would like to have this feature added, but closing due to no response.

**Pull request closed without merging**